rs/src/management/http_client.rs

```diff
@@ -0,0 +1,560 @@
+    pub async fn update_tunnel_port(
+        &self,
+        locator: &TunnelLocator,
+        port: &TunnelPort,
+        options: &TunnelRequestOptions,
+    ) -> HttpResult<TunnelPort> {
+        let url = self.build_tunnel_uri(
+            locator,
+            Some(&format!("{}/{}", PORTS_API_SUB_PATH, port.port_number)),
+        );
+        let mut request = self.make_tunnel_request(Method::PUT, url, options);
+        json_body(&mut request, port);
+        self.execute_json("create_tunnel_port", request).await
+    }
+
+    /// Updates an existing port on the tunnel.
+    pub async fn delete_tunnel_port(
+        &self,
+        locator: &TunnelLocator,
+        port_number: u16,
+        options: &TunnelRequestOptions,
+    ) -> HttpResult<TunnelPort> {
+        let url = self.build_tunnel_uri(
+            locator,
+            Some(&format!("{}/{}", PORTS_API_SUB_PATH, port_number)),
+        );
+        let request = self.make_tunnel_request(Method::DELETE, url, options);
+        self.execute_json("create_tunnel_port", request).await
+    }
+
+    /// Sends the request and deserializes a JSON response
+    #[cfg(feature = "instrumentation")]
+    async fn execute_json<T>(&self, feature: &'static str, request: Request) -> HttpResult<T>
+    where
+        T: DeserializeOwned,
+    {
+        use opentelemetry::{
+            global,
+            trace::{TraceContextExt, Tracer},
+        };
+
+        let tracer = global::tracer("tunneling");
+        let span = tracer.start(feature);
+        let cx = opentelemetry::Context::current_with_span(span);
+        let guard = cx.clone().attach();
+
+        let res = self.execute_json_simple(request).await;
+        if let Err(e) = &res {
+            cx.span().record_exception(e);
+        }
+
+        drop(guard);
+
+        res
+    }
+    /// Sends the request and deserializes a JSON response
+    #[cfg(not(feature = "instrumentation"))]
+    async fn execute_json<T>(&self, _: &'static str, request: Request) -> HttpResult<T>
+    where
+        T: DeserializeOwned,
+    {
+        self.execute_json_simple(request).await
+    }
+
+    async fn execute_json_simple<T>(&self, request: Request) -> HttpResult<T>
+    where
+        T: DeserializeOwned,
+    {
+        let url_clone = request.url().clone();
+        let res = self
+            .client
+            .execute(request)
+            .await
+            .map_err(HttpError::ConnectionError)?;
+
+        if res.status().is_success() {
+            res.json::<T>().await.map_err(HttpError::ConnectionError)
+        } else {
+            let request_id = res
+                .headers()
+                .get("VsSaaS-Request-Id")
+                .and_then(|h| h.to_str().ok())
+                .map(|s| s.to_owned());
+
+            Err(HttpError::ResponseError(ResponseError {
+                url: url_clone,
+                status_code: res.status(),
+                data: res.text().await.ok(),
```

Comment:
Most error responses should include a JSON `ProblemDetails` object. Should that be deserialized here?